vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/DatabaseWorkflow.java

```diff
@@ -192,11 +192,12 @@ public static DatabaseWorkflow reinitialise(
       AtomicBoolean liveness,
       Set<ExternalKey> keys,
       Map<String, JsonNode> consumableResources,
-      DSLContext dsl)
+      DSLContext dsl,
+      int attempt)
       throws SQLException {
-    final var attempt =
-        dsl.update(ACTIVE_WORKFLOW_RUN)
-            .set(ACTIVE_WORKFLOW_RUN.ATTEMPT, ACTIVE_WORKFLOW_RUN.ATTEMPT.plus(1))
+
+    dsl.update(ACTIVE_WORKFLOW_RUN)
+            .set(ACTIVE_WORKFLOW_RUN.ATTEMPT, attempt)
```

Comment:
this would allow someone to submit a job with a new priority and an attempt number of 50000 and we'd just accept it as the truth

Reply:
since the check at DatabaseBackedProcessor:1119 enforces a reasonable attempt count and we skip that part of the WHERE since there's an OR now

Reply:
Mike and I had discussed this earlier and he suggested it was minimal risk, but I can change this to take into account attempt and priority separately (will just be duplicated code)

Reply:
This change is not minimal risk and absolutely should be reverted. The purpose of the attempt number is to provide distributed transaction coherency between all the actors involved. It is clearly that the purpose of this is not obvious, so I will elaborate.

Vidarr has a workflow run and then something, probably Shesmu or maybe a human does a read on that data. Now, suppose two Shesmu actions (or a human) want to update this data. They increment the attempt number by one. If they both write, one will win and the other will lose and the loser can be aware that they attempted a write on stale data and they have to read again and reattempt (if desired).

You'll not that on `DatabaseBackedProcessor`, the comparison on 1102 is `ACTIVE_WORKFLOW_RUN.ATTEMPT.eq(attempt - 1)`, to avoid letting someone punch in a large value to override the current state of the database.

Ultimately, the Vidarr database has to be the single source of the truth. Trusting the external parties to write to this field directly utterly destroys the value of this already weak transaction mechanism.

Reply:
I believe the addition of `.and(ACTIVE_WORKFLOW_RUN.ATTEMPT.eq(attempt))` on 1131  should provide an equivalent safeguard as the comparison on 1102 which will avoid letting someone punch in a large value to override the current state of the database.

Reply:
That's even less reason to have this change.